File: external/cpp-impl/src/structures/entity.cpp
//
// Created by jldevezas on 3/8/18.
//

#include <structures/entity.h>

Entity::Entity(std::string label, std::string uri) {
    this->label = label;
    this->uri = uri;
}

const std::string &Entity::getLabel() const {
    return label;
}

void Entity::setLabel(const std::string &label) {
    Entity::label = label;
}

const std::string &Entity::getUri() const {
    return uri;
}

void Entity::setUri(const std::string &uri) {
    Entity::uri = uri;
}

File: external/java-impl/src/main/java/armyant/structures/Entity.java
package armyant.structures;

import java.util.Objects;

public class Entity {
    private String uri;
    private String label;

    public Entity(String label) {
        this(label, null);
    }

    public Entity(String label, String uri) {
        this.uri = uri;
        this.label = label;
    }

    public String getURI() {
        return uri;
    }

    public void setURI(String uri) {
        this.uri = uri;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    @Override
    public boolean equals(Object o) {
        if (o == this)
            return true;
        if (!(o instanceof Entity)) {
            return false;
        }
        Entity entity = (Entity) o;
        return Objects.equals(uri, entity.uri);
    }

    @Override
    public int hashCode() {
        return Objects.hash(uri);
    }
}